.github/workflows/test.yml

```diff
@@ -32,5 +32,16 @@ jobs:
       with:
         image: ghcr.io/osbuild/osbuild-ci:latest-202308241910
         run: |
+          # Note that only "test.run.test_stages" runs in parallel because
+          # the other tests are not sufficiently isolated and will cause
+          # random failures. But test_stages is the long running one with
+          # almost 2h.
+          if [ "${{ matrix.test }}" = "test.run.test_stages" ]; then
+              # Using 4 workers is a bit arbitrary, "auto" is probably too
+              # aggressive.
+              export TEST_WORKERS="-n 4"
+              # Share the store between the workers speeds things up further
```

Comment:
Fwiw, I'm not 100% certain how much effect this has in GH, I saw speedups in my local runs from 41min to 31min with it) but have not measured this as well GH.